src/descriptors.test.ts

```diff
@@ -173,3 +174,191 @@ describe("getChecksum", () => {
+
+  const expectedRangeKeys = [
+    {
+      xfp: "d52d08fc",
+      bip32Path: "m/48h/1h/0h/2h",
+      xpub: "tpubDEmdqnW7FVtXFej7WqNH6Wt92LKECvi2C326HHziBbfq8XCqS1qVoWYMpzPZcAMoD5n4YBDDdkRSToZsP3fgJSzMkLexZ6M3Vsuw7aXdZtz",
+    },
+    {
+      xfp: "85b4d568",
+      bip32Path: "m/48h/1h/0h/2h",
+      xpub: "tpubDFg79ktERPWQb7L8BFJFhCWq3hrZfGKz393LmB9eXgAg9TLh1GgPSa6XD5TyWrKkSUkijwajoMHQc4yRNwUqsoyC7sW4tb1EutYBfEm1boX",
+    },
+  ];
+
+  // Expected individual descriptors after expansion
+  const expectedExpandedExternal =
+    "wsh(sortedmulti(2,[d52d08fc/48h/1h/0h/2h]tpubDEmdqnW7FVtXFej7WqNH6Wt92LKECvi2C326HHziBbfq8XCqS1qVoWYMpzPZcAMoD5n4YBDDdkRSToZsP3fgJSzMkLexZ6M3Vsuw7aXdZtz/0/*,[85b4d568/48h/1h/0h/2h]tpubDFg79ktERPWQb7L8BFJFhCWq3hrZfGKz393LmB9eXgAg9TLh1GgPSa6XD5TyWrKkSUkijwajoMHQc4yRNwUqsoyC7sW4tb1EutYBfEm1boX/0/*))";
+  const expectedExpandedInternal =
+    "wsh(sortedmulti(2,[d52d08fc/48h/1h/0h/2h]tpubDEmdqnW7FVtXFej7WqNH6Wt92LKECvi2C326HHziBbfq8XCqS1qVoWYMpzPZcAMoD5n4YBDDdkRSToZsP3fgJSzMkLexZ6M3Vsuw7aXdZtz/1/*,[85b4d568/48h/1h/0h/2h]tpubDFg79ktERPWQb7L8BFJFhCWq3hrZfGKz393LmB9eXgAg9TLh1GgPSa6XD5TyWrKkSUkijwajoMHQc4yRNwUqsoyC7sW4tb1EutYBfEm1boX/1/*))";
+
+  describe("getWalletFromDescriptor with range notation", () => {
+    it("should decode descriptor with <0;1> range notation", async () => {
+      const config = await getWalletFromDescriptor(
+        rangeDescriptor,
+        Network.TESTNET,
+      );
+
+      expect(config.addressType).toEqual("P2WSH");
+      expect(config.requiredSigners).toEqual(2);
+      expect(config.network).toEqual(Network.TESTNET);
+      expect(config.keyOrigins.length).toEqual(2);
+
+      // Verify first key origin
+      expect(config.keyOrigins[0].xfp).toEqual(expectedRangeKeys[0].xfp);
+      expect(config.keyOrigins[0].xpub).toEqual(expectedRangeKeys[0].xpub);
+
+      // Verify second key origin
+      expect(config.keyOrigins[1].xfp).toEqual(expectedRangeKeys[1].xfp);
+      expect(config.keyOrigins[1].xpub).toEqual(expectedRangeKeys[1].xpub);
+    });
+
+    it("should handle range notation with different formats", async () => {
+      // Test with spaces in range
+      const rangeWithSpaces =
+        "wsh(sortedmulti(2,[d52d08fc/48h/1h/0h/2h]tpubDEmdqnW7FVtXFej7WqNH6Wt92LKECvi2C326HHziBbfq8XCqS1qVoWYMpzPZcAMoD5n4YBDDdkRSToZsP3fgJSzMkLexZ6M3Vsuw7aXdZtz/< 0 ; 1 >/*,[85b4d568/48h/1h/0h/2h]tpubDFg79ktERPWQb7L8BFJFhCWq3hrZfGKz393LmB9eXgAg9TLh1GgPSa6XD5TyWrKkSUkijwajoMHQc4yRNwUqsoyC7sW4tb1EutYBfEm1boX/< 0 ; 1 >/*))#pjv8pr5k";
+
+      const config = await getWalletFromDescriptor(
+        rangeWithSpaces,
+        Network.TESTNET,
+      );
+      expect(config.addressType).toEqual("P2WSH");
+      expect(config.requiredSigners).toEqual(2);
```

Comment:
can we do all the same checks as in the previous test? you could even make this a set of vectors [to loop through](https://jestjs.io/docs/api#each)